src/ImageSearchModal.jsx

```diff
@@ -15,238 +15,203 @@ import cockpit from 'cockpit';
 import rest from './rest.js';
 import * as client from './client.js';
 import { fallbackRegistries } from './util.js';
-import { DialogsContext } from "dialogs.jsx";
+import { useDialogs } from "dialogs.jsx";
 
 import './ImageSearchModal.css';
 
 const _ = cockpit.gettext;
 
-export class ImageSearchModal extends React.Component {
-    static contextType = DialogsContext;
-
-    constructor(props) {
-        super(props);
-        this.state = {
-            imageIdentifier: '',
-            imageList: [],
-            searchInProgress: false,
-            searchFinished: false,
-            isSystem: props.systemServiceAvailable,
-            registry: "",
-        };
-        this.onDownloadClicked = this.onDownloadClicked.bind(this);
-        this.onItemSelected = this.onItemSelected.bind(this);
-        this.onSearchTriggered = this.onSearchTriggered.bind(this);
-        this.onValueChanged = this.onValueChanged.bind(this);
-        this.onKeyPress = this.onKeyPress.bind(this);
-        this.onToggleUser = this.onToggleUser.bind(this);
-    }
-
-    componentDidMount() {
-        this._isMounted = true;
-    }
-
-    componentWillUnmount() {
-        this._isMounted = false;
-
-        if (this.activeConnection)
-            this.activeConnection.close();
-    }
-
-    onToggleUser(ev) {
-        this.setState({ isSystem: ev.currentTarget.value === "system" });
-    }
-
-    onDownloadClicked() {
-        const Dialogs = this.context;
-        const selectedImageName = this.state.imageList[this.state.selected].Name;
-
-        Dialogs.close();
-        this.props.downloadImage(selectedImageName, this.state.imageTag, this.state.isSystem);
-    }
-
-    onItemSelected(_, key) {
-        this.setState({ selected: key.split('-').slice(-1)[0] });
-    }
-
-    onSearchTriggered(forceSearch) {
+export const ImageSearchModal = ({ downloadImage, registries, user, userServiceAvailable, systemServiceAvailable }) => {
+    const [searchInProgress, setSearchInProgress] = useState(false);
+    const [searchFinished, setSearchFinished] = useState(false);
+    const [imageIdentifier, setImageIdentifier] = useState('');
+    const [imageList, setImageList] = useState([]);
+    const [imageTag, setImageTag] = useState("");
+    const [isSystem, setIsSystem] = useState(systemServiceAvailable);
+    const [selectedRegistry, setSelectedRegistry] = useState("");
+    const [selected, setSelected] = useState("");
+    const [dialogError, setDialogError] = useState("");
+    const [dialogErrorDetail, setDialogErrorDetail] = useState("");
+    const [typingTimeout, setTypingTimeout] = useState(null);
+
+    let activeConnection = null;
+    const Dialogs = useDialogs();
+    // Registries to use for searching
+    const searchRegistries = registries.search && registries.length !== 0 ? registries.search : fallbackRegistries;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1332-20230705-092041-36f39867-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)